magda-web-client/src/helpers/ChartDatasetEncoder.js

```diff
@@ -123,6 +222,147 @@ const defaultChartOption = {
     color: ["#8856a7", "#9ebcda", "#e0ecf4"]
 };
 
+function testKeywords(str, keywords) {
+    if (!str) {
+        return false;
+    } else if (!keywords || !keywords.length) {
+        return false;
+    } else {
+        const r = new RegExp(`.*(${keywords.join("|")}).*`, "i");
+        return !!r.test(str);
+    }
+}
+
+function fieldDefAdjustment(field) {
+    if (testKeywords(field.label, OVERRIDE_TO_CATEGORY_COLUMNS)) {
+        return {
+            ...field,
+            numeric: false,
+            category: true,
+            time: false
+        };
+    } else if (testKeywords(field.label, OVERRIDE_TO_TIME_COLUMNS)) {
+        field.numeric = false;
+        field.category = false;
+        field.time = true;
+    } else if (testKeywords(field.label, OVERRIDE_TO_NUMERIC_COLUMNS)) {
+        field.numeric = true;
+        field.category = false;
+        field.time = false;
+    }
+    return field;
+}
+
+function preProcessFields(headerRow, distribution) {
+    let disFields =
+        distribution.visualizationInfo && distribution.visualizationInfo.fields;
+    if (!disFields) disFields = [];
+
+    let newFields = map(disFields, (field, key) => ({
+        ...field,
+        idx: indexOf(headerRow, key),
+        name: key,
+        label: startCase(key),
+        category: !field.time && !field.numeric,
+        isAggr: false
+    }));
+    //--- filter out fields that cannot be located in CSV data. VisualInfo outdated maybe?
+    newFields = filter(disFields, item => item.idx !== -1);
```

Comment:
This dataset now shows a blank column (in the xAxis dropdown):
https://1140-preview-chart-dates.dev.magda.io/dataset/ds-sa-5db51968-5902-4bf5-a177-ad83d805bd38/details?q=income
But production site looks ok:
https://search.data.gov.au/dataset/ds-sa-5db51968-5902-4bf5-a177-ad83d805bd38/details?q=income
I think the issue might be around this line (`disFields` should be `newFields`) but couldn't be certain

Reply:
Yeah oops 😬